transformations/urban_dict_swap/transformation.py

```diff
@@ -0,0 +1,77 @@
+import random
+
+import requests
+import spacy
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+
+from initialize import spacy_nlp
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+
+class UrbanThesaurusSwap(SentenceOperation):
+    """
+    Randomly swap nouns from the input text for related terms from the Urban Dictionary.
+    """
+
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+        TaskType.TEXT_TAGGING,
+    ]
+    languages = ["en"]
+    keywords = ["lexical", "api-based", "unnaturally-written", "possible-meaning-alteration", "high-precision",
+                "high-coverage", "figurative-language"]
+
+    retry_strategy = Retry(
+        total=3,
+        status_forcelist=[429, 500, 502, 503, 504],
+        method_whitelist=["HEAD", "GET", "OPTIONS"],
+    )
+    adapter = HTTPAdapter(max_retries=retry_strategy)
+    http = requests.Session()
+    http.mount("https://", adapter)
+
+    def __init__(self, seed=0, prob=0.5, min_score=100.0, max_outputs=1):
+        super().__init__(seed, max_outputs=max_outputs)
+        self.seed = seed
+        self.nlp = spacy_nlp if spacy_nlp else spacy.load("en_core_web_sm")
+        self.prob = prob
+        self.min_score = min_score
+        self.max_outputs = max_outputs
+        self.url = "https://urbanthesaurus.org/api/related?term="
```

Comment:
Aren't there APIs for synonymy relationships? Might be specific as compared to "related" ?

Reply:
Also can the "min_score" be used to reduce the size of the candidates called? UrbanDictionary is a pretty large dictionary.

Reply:
Unfortunately, the API is undocumented, and I haven't been able to discover any other filter params. The official Urban Dictionary API doesn't support related terms/synonymy searching, just definitions.